ui/app/src/views/home/HomeView.tsx

```diff
@@ -0,0 +1,148 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Box, Button, Card, Grid, Stack, Typography } from '@mui/material';
+import { useCallback, useMemo, useState } from 'react';
+import HomeIcon from 'mdi-material-ui/Home';
+import { useNavigate } from 'react-router-dom';
+import HistoryIcon from 'mdi-material-ui/History';
+import FormatListBulletedIcon from 'mdi-material-ui/FormatListBulleted';
+import ViewDashboardIcon from 'mdi-material-ui/ViewDashboard';
+import { DashboardSelector } from '@perses-dev/core';
+import { ProjectModel, useProjectList } from '../../model/project-client';
+import { AddProjectDialog } from '../../components/AddProjectDialog/AddProjectDialog';
+import { CreateDashboardDialog } from '../../components/CreateDashboardDialog/CreateDashboardDialog';
+import DashboardBreadcrumbs from '../../components/DashboardBreadcrumbs';
+import { InformationSection } from './InformationSection';
+import { RecentDashboardsMosaic } from './RecentDashboardsMosaic';
+import { SearchableDashboards } from './SearchableDashboards';
+import { ImportantDashboard } from './ImportantDashboards';
+
+function HomeView() {
+  // Navigate to the project page if the project has been successfully added
+  const navigate = useNavigate();
+
+  const { data } = useProjectList();
+
+  const projectOptions = useMemo(() => {
+    return (data || []).map((project) => project.metadata.name);
+  }, [data]);
+
+  const handleAddProjectDialogSubmit = useCallback(
+    (entity: ProjectModel) => navigate(`/projects/${entity.metadata.name}`),
+    [navigate]
+  );
+
+  const handleAddDashboardDialogSubmit = useCallback(
+    (dashboardSelector: DashboardSelector) =>
+      navigate(`/projects/${dashboardSelector.project}/dashboards/${dashboardSelector.dashboard}/create`),
+    [navigate]
+  );
+
+  // Open/Close management for dialogs
+  const [isAddProjectDialogOpen, setIsAddProjectDialogOpen] = useState(false);
+  const [isAddDashboardDialogOpen, setIsAddDashboardDialogOpen] = useState(false);
+
+  const handleAddProjectDialogOpen = useCallback(() => {
```

Comment:
IMO i think it's ok to skip `useCallback` for simple functions like these. I don't think it'll be too much of a performance impact